### ISSUE TITLE
[java] UnusedPrivateMethod Call of private method with implementation as parameter

### ISSUE THREAD

**Issue by osek666** [pmd/pmd#777]
**Rule Set:**
Unused Code

**Rule:**
UnusedPrivateMethod

**Description:**

***Given***
Interface `A`
Class `B` that implements `A`
Class `C` with 
- private method `m`, that takes `A` as parameter
- public method `n` that calls `m` with instance of `B` as parameter

***Result*** 
PMD reports method `m` as unused

**Code Sample demonstrating the issue:**

```java
public interface A {}
```

```java
public class B implements A {}
```

```java
public class C {

    public void n() {
        m(new B());
    }

    private void m(final A param) {
        System.out.println("C.m");
        System.out.println("param = [" + param + "]");
    }
}
```

***Result***
```
PMD Failure: C:7 Rule:UnusedPrivateMethod Priority:3 Avoid unused private methods such as 'm(A)'..
```

**Running PMD through:** *Maven*

**Comment by jsotuyod** [comment]
@osek666 thanks for the report. This seems to be a duplicate of #770. I'm closing this to keep tracking it there.
